frame/proxy/src/lib.rs

```diff
@@ -800,4 +800,18 @@ impl<T: Config> Pallet<T> {
 		let e = call.dispatch(origin);
 		Self::deposit_event(Event::ProxyExecuted { result: e.map(|_| ()).map_err(|e| e.error) });
 	}
+
+	fn remove_all_proxy_delegates(who: &T::AccountId) {
+		<Proxies<T>>::mutate(| proxies: &mut <Proxies<T>> | 
```

Comment:
Why does the body of this function look so different from the code that you removed from the extrinsic?

Reply:
I was getting many errors on previous code after I removed the trait implementation such as proxies.remove method not found in. So I was trying different approach. It also has a bug to fix. Their are some namespace conflicts. Shortly i am going to push the fixed one

Reply:
I'm unsure what kind of bug fix you've made, but one glaring behaviour change needs to be explained. Why is there a `delegates.clear()` call?

Reply:
What kind of bug fix? In any case, the code changes for it doesn't belong to this PR, so please revert it.

Reply:
reverted